src/AspNet.Security.OAuth.Slack/SlackAuthenticationHelper.cs

```diff
@@ -17,133 +17,89 @@ namespace AspNet.Security.OAuth.Slack
     public static class SlackAuthenticationHelper
     {
         /// <summary>
-        /// Gets the identifier corresponding to the authenticated user.
-        /// </summary>
-        public static string GetUserIdentifier([NotNull] JObject user)
-        {
-            if (user == null)
-            {
-                throw new ArgumentNullException(nameof(user));
-            }
-
-            return user.Value<string>("user_id");
-        }
-
-        /// <summary>
-        /// Gets the login corresponding to the authenticated user.
+        /// Returns a unique identifer for the authenticated user.
         /// </summary>
-        public static string GetUserName([NotNull] JObject user)
+        /// <remarks>
+        /// According to the Slack API documentation at https://api.slack.com/methods/users.identity, user IDs are
+        /// not guaranteed to be globally unique across all Slack users.  The combination of user ID and team ID, 
+        /// on the other hand, is guaranteed to be globally unique.
+        /// </remarks>
+        public static string GetUniqueIdentifier([NotNull] JObject user)
         {
             if (user == null)
             {
                 throw new ArgumentNullException(nameof(user));
             }
 
-            return user.Value<string>("user");
-        }
-
-        /// <summary>
-        /// Gets the name corresponding to the authenticated user.
-        /// </summary>
-        public static string GetTeamIdentifier([NotNull] JObject user)
-        {
-            if (user == null)
-            {
-                throw new ArgumentNullException(nameof(user));
-            }
+            string userId = user.Value<JObject>("user")?.Value<string>("id");
+            string teamId = user.Value<JObject>("team")?.Value<string>("id");
 
-            return user.Value<string>("team_id");
+            return $"{teamId}|{userId}";
```

Comment:
Could we simply do `return $"{GetTeamIdentifier(user)}|{GetUserIdentifier(user)}";` and avoid the two intermediate variables?